### PR TITLE
Eugene work

### PR DESCRIPTION
**Pull request by edtyler**
### Key features
Add SetName for Restaurant, set/get Administrator for User, FindbyNumber for TripDataStore
Added GetNumber to MenuItem
Added TripDataStore::StoreTripNumRest() and TripDataStore::StoreTrip(). see dbtest.cpp for usage

### Notes
see dbtest.cpp for usage and unit tests

### CODE REVIEW

File: src/datastore/ExtraRestaurantData.csv [hunk 1]
```diff
@@ -0,0 +1,2 @@
+11,Pizza Hut,10.1,0,0,0,0,12,1,11.5,2,14.2,3,13.4,4,16.5,5,3.3,6,12.2,7,7.8,8,6.9,9,8.2,10,6.11,11,0.0,12,5.8,6,1,0,Veggie pizza,4.99,2,0,Pepperoni pizza,8.99,3,0,All meat pizza,12.99,4,0,Cheese pizza,5.99,5,0,Mushroom and pepperoni pizza,6.99,6,0,Mushroom and green pepper pizza,7.29,0,
```
Comment (typokign) [inline src/datastore/ExtraRestaurantData.csv:1]: plz capitalize menu items

File: src/datastore/ExtraRestaurantData.csv [hunk 2]
```diff
@@ -0,0 +1,2 @@
...
+12,Sonic,6.6,0,0,0,0,12,1,13.8,2,4.4,3,18.1,4,18.8,5,0.8,6,8.2,7,10.8,8,9.4,9,5.5,10,11.8,11,5.8,12,0.0,5,1,0,Sonic hamburger,4.99,2,0,Sonic cheeseburger,4.29,3,0,New York dog,2.89,4,0,French fries,1.79,5,0,Medium Pepsi,2.55,0
```
Comment (typokign) [inline src/datastore/ExtraRestaurantData.csv:2]: same here

File: src/datastore/ExtraStacksRestaurantData.csv [hunk 3]
```diff
@@ -0,0 +1 @@
+13,Stacks,10.1,0,0,0,0,13,1,11.5,2,14.2,3,13.4,4,16.5,5,3.3,6,12.2,7,7.8,8,6.9,9,8.2,10,6.11,11,5.1,12,5.8,13,0,3,1,0,Avacado Omlette,8.99,2,0,Macadami Nut Waffle,12.99,3,0,Spam and Eggs,2.99,0,
```
Comment (typokign) [inline src/datastore/ExtraStacksRestaurantData.csv:1]: good capitalization, thx, but:

s/avacado/avocado
s/omlette/omelette
s/macadami/macadamia

File: src/datastore/MenuItem.cpp [hunk 4]
```diff
@@ -17,6 +17,11 @@ MenuItem::~MenuItem()
 {
 }
 
+int MenuItem::GetNumber(void) const
```
Comment (typokign) [inline src/datastore/MenuItem.cpp:20]: camelCase methods please

File: src/datastore/Restaurant.cpp [hunk 5]
```diff
@@ -96,6 +96,11 @@ const string &Restaurant::GetName(void) const
     return m_Name;
 }
 
+void Restaurant::SetName(const string &NewName)
```
Comment (typokign) [inline src/datastore/Restaurant.cpp:99]: camelCase methods and parameters please

File: src/datastore/RestaurantDataStore.cpp [hunk 6]
```diff
@@ -22,10 +22,10 @@ RestaurantDataStore::RestaurantDataStore()
 {
 };
 
-void RestaurantDataStore::load(const string path)
+void RestaurantDataStore::load(const string path, bool ItemsAreAdditional)
```
Comment (typokign) [inline src/datastore/RestaurantDataStore.cpp:25]: camelCase

File: src/datastore/RestaurantDataStore.cpp [hunk 7]
```diff
@@ -22,10 +22,10 @@ RestaurantDataStore::RestaurantDataStore()
 {
 };
 
-void RestaurantDataStore::load(const string path)
+void RestaurantDataStore::load(const string path, bool ItemsAreAdditional)
 {
-    string fullpath = path;
-    std::ifstream infile(fullpath, ios::in);
+    //string fullpath = path + "RestaurantData.csv";
```
Comment (typokign) [inline src/datastore/RestaurantDataStore.cpp:27]: cruft

File: src/datastore/RestaurantDataStore.cpp [hunk 8]
```diff
@@ -133,11 +166,11 @@ void RestaurantDataStore::load(const string path)
 
 void RestaurantDataStore::save(const string path)
 {
-    string fullpath = path + ".tmp";
+    //string fullpath = path + "RestaurantData.csv.tmp";
```
Comment (typokign) [inline src/datastore/RestaurantDataStore.cpp:169]: cruft

File: src/datastore/RestaurantDataStore.cpp [hunk 9]
```diff
@@ -202,6 +235,11 @@ void RestaurantDataStore::save(const string path)
     }
 }
 
+void RestaurantDataStore::load_additional(const string path)
```
Comment (typokign) [inline src/datastore/RestaurantDataStore.cpp:238]: camelCase please

File: src/datastore/RestaurantDataStore.cpp [hunk 10]
```diff
@@ -214,9 +252,24 @@ Restaurant &RestaurantDataStore::FindbyNumber(int Number)
     return *(list.end());  // never reached  - should throw exception
 }
 
+bool RestaurantDataStore::DuplicateNumPresent(int Number)
```
Comment (typokign) [inline src/datastore/RestaurantDataStore.cpp:255]: camelCase method and param please

File: src/datastore/RestaurantDataStore.cpp [hunk 11]
```diff
@@ -214,9 +252,24 @@ Restaurant &RestaurantDataStore::FindbyNumber(int Number)
     return *(list.end());  // never reached  - should throw exception
 }
 
+bool RestaurantDataStore::DuplicateNumPresent(int Number)
+{
+    bool dupe_found = false;
```
Comment (typokign) [inline src/datastore/RestaurantDataStore.cpp:257]: camelCase variables

File: src/datastore/RestaurantDataStore.cpp [hunk 12]
```diff
@@ -214,9 +252,24 @@ Restaurant &RestaurantDataStore::FindbyNumber(int Number)
...
+        }
+    }
+    return dupe_found;
+}
+
 void RestaurantDataStore::printAsDebug(bool printeol, bool printcontent) const
 {
-//    list.printAsDebug(printeol,printcontent);
+    // std::list has no printAsDebug() method - MyDblLinkList does
+    //list.printAsDebug(printeol,printcontent);
```
Comment (typokign) [inline src/datastore/RestaurantDataStore.cpp:272]: cruft

File: src/datastore/RestaurantDataStore.hpp [hunk 13]
```diff
@@ -29,13 +29,15 @@ class RestaurantDataStore
     void printAsDebug(bool printeol, bool printcontent) const;
     Restaurant &FindbyNumber(int Number);
 
-    void load(const string path);
+    void load(const string path, bool ItemsAreAdditional = false);
```
Comment (typokign) [inline src/datastore/RestaurantDataStore.hpp:32]: camelCase

File: src/datastore/RestaurantDataStore.hpp [hunk 14]
```diff
@@ -29,13 +29,15 @@ class RestaurantDataStore
     void printAsDebug(bool printeol, bool printcontent) const;
     Restaurant &FindbyNumber(int Number);
 
-    void load(const string path);
+    void load(const string path, bool ItemsAreAdditional = false);
     void save(const string path);
+    void load_additional(const string path);
```
Comment (typokign) [inline src/datastore/RestaurantDataStore.hpp:34]: camelCase

File: src/datastore/RestaurantDataStore.hpp [hunk 15]
```diff
@@ -29,13 +29,15 @@ class RestaurantDataStore
...
-    void load(const string path);
+    void load(const string path, bool ItemsAreAdditional = false);
     void save(const string path);
+    void load_additional(const string path);
 
     std::list<Restaurant> list;
 //    MyDblLinkList<Restaurant> list;
 
 private:
+    bool DuplicateNumPresent(int Number);
```
Comment (typokign) [inline src/datastore/RestaurantDataStore.hpp:40]: camelCase

File: src/datastore/Trip.cpp [hunk 16]
```diff
@@ -93,6 +93,10 @@ Trip& Trip::operator=(const Trip& rhs)
    return *this;
 }
 
+int Trip::GetNumber(void) const
```
Comment (typokign) [inline src/datastore/Trip.cpp:96]: camelCase, and remove `void` please

File: src/datastore/Trip.hpp [hunk 17]
```diff
@@ -54,6 +54,7 @@ friend class TripDataStore;
     // Assignment operator
     Trip& operator=(const Trip& rhs);
 
+    int GetNumber(void) const;
```
Comment (typokign) [inline src/datastore/Trip.hpp:57]: camelCase and void

File: src/datastore/TripDataStore.cpp [hunk 18]
```diff
@@ -25,8 +28,8 @@ TripDataStore::TripDataStore()
 
 void TripDataStore::load(const string path)
 {
-    string fullpath = path + "TripData.csv";
-    std::ifstream infile(fullpath, ios::in);
+    //string fullpath = path + "TripData.csv";
```
Comment (typokign) [inline src/datastore/TripDataStore.cpp:31]: cruft

File: src/datastore/TripDataStore.cpp [hunk 19]
```diff
@@ -132,6 +143,218 @@ void TripDataStore::save(const string path)
 
 }
 
+Trip &TripDataStore::FindbyNumber(int Number)
```
Comment (typokign) [inline src/datastore/TripDataStore.cpp:146]: camelCase

File: src/datastore/TripDataStore.cpp [hunk 20]
```diff
@@ -132,6 +143,218 @@ void TripDataStore::save(const string path)
...
+    {
+        if ( (*it).m_nNumber == Number)
+        {
+            return *it;
+        }
+    }
+    return *(list.end());  // never reached  - should throw exception
+}
+
+bool TripDataStore::DuplicateNumPresent(int Number)
```
Comment (typokign) [inline src/datastore/TripDataStore.cpp:158]: camelCase

File: src/datastore/TripDataStore.cpp [hunk 21]
```diff
@@ -132,6 +143,218 @@ void TripDataStore::save(const string path)
...
+{
+    bool dupe_found = false;
```
Comment (typokign) [inline src/datastore/TripDataStore.cpp:160]: camelCase

File: src/datastore/TripDataStore.cpp [hunk 22]
```diff
@@ -132,6 +143,218 @@ void TripDataStore::save(const string path)
...
+            dupe_found = true;
+            break;
+        }
+    }
+    return dupe_found;
+}
+
+int TripDataStore::StoreTrip(const string &TripName, const vector<int> RestaurantsSelectedbyUser, RestaurantDataStore &RestSt, User &User, bool StartatSaddleback)
+{
+    int current_restaurant;
```
Comment (typokign) [inline src/datastore/TripDataStore.cpp:174]: camelCase 174:180

File: src/datastore/TripDataStore.cpp [hunk 23]
```diff
@@ -132,6 +143,218 @@ void TripDataStore::save(const string path)
...
+        if ( (*it).m_nNumber == Number)
+        {
+            dupe_found = true;
+            break;
+        }
+    }
+    return dupe_found;
+}
+
+int TripDataStore::StoreTrip(const string &TripName, const vector<int> RestaurantsSelectedbyUser, RestaurantDataStore &RestSt, User &User, bool StartatSaddleback)
```
Comment (typokign) [inline src/datastore/TripDataStore.cpp:172]: camelCase

File: src/datastore/TripDataStore.cpp [hunk 24]
```diff
@@ -132,6 +143,218 @@ void TripDataStore::save(const string path)
...
+    float shortest_distance = 999.9;
+    float cumulative_distance = 0;
+    Restaurant *pRest;
+    vector<int> selected_restaurants(RestaurantsSelectedbyUser.begin(), RestaurantsSelectedbyUser.end());
+    if (StartatSaddleback)
+    {
+        for (vector<int>::const_iterator it = selected_restaurants.begin(); it != selected_restaurants.end(); it++)
+        {
+            pRest = &RestSt.FindbyNumber(*it);
+            float restdist = pRest->GetDistSaddleback();
```
Comment (typokign) [inline src/datastore/TripDataStore.cpp:186]: camelCase

File: src/datastore/TripDataStore.cpp [hunk 25]
```diff
@@ -132,6 +143,218 @@ void TripDataStore::save(const string path)
...
+        pRest->AddTrip(tmp.GetNumber());
+    }
+
+    // also update the creating user's list
+    User.AddTrip(tmp.GetNumber());
+
+    return(tmp.GetNumber());
+}
+
+int TripDataStore::StoreTripNumRest(const string &TripName, int StartingRestNum, int NumtoVisit, RestaurantDataStore &RestSt, User &User)
```
Comment (typokign) [inline src/datastore/TripDataStore.cpp:276]: camelCase

File: src/datastore/TripDataStore.cpp [hunk 26]
```diff
@@ -132,6 +143,218 @@ void TripDataStore::save(const string path)
...
+{
+    int current_restaurant = StartingRestNum;
```
Comment (typokign) [inline src/datastore/TripDataStore.cpp:278]: camelCase 278:282

File: src/datastore/TripDataStore.cpp [hunk 27]
```diff
@@ -132,6 +143,218 @@ void TripDataStore::save(const string path)
...
+    for (vector<int>::iterator it = sVec.begin(); it != sVec.end(); it++)
+    {
+        pRest = &RestSt.FindbyNumber(*it);
+        pRest->AddTrip(tmp.GetNumber());
+    }
+
+    // also update the creating user's list
+    User.AddTrip(tmp.GetNumber());
+
+    return(tmp.GetNumber());
```
Comment (typokign) [inline src/datastore/TripDataStore.cpp:355]: unnecessary parentheses?

File: src/datastore/User.hpp [hunk 28]
```diff
@@ -24,7 +24,7 @@ friend class UserDataStore;
     friend ostream& operator<<(ostream& os, const User& user);
 
     // No arg constructor
-    User() { m_nUserNumber = -1; m_bInitialized = false; };
+    User() { m_nNumber = -1; m_bInitialized = false; };
```
Comment (typokign) [inline src/datastore/User.hpp:27]: break this out into two lines please?

File: src/datastore/UserDataStore.cpp [hunk 29]
```diff
@@ -26,8 +26,8 @@ UserDataStore::UserDataStore()
 
 void UserDataStore::load(const string path)
 {
-    string fullpath = path + "UserData.csv";
-    std::ifstream infile(fullpath, ios::in);
+    //string fullpath = path + "UserData.csv";
```
Comment (typokign) [inline src/datastore/UserDataStore.cpp:29]: cruft

File: src/datastore/UserDataStore.cpp [hunk 30]
```diff
@@ -93,18 +102,18 @@ void UserDataStore::load(const string path)
 
 void UserDataStore::save(const string path)
 {
-    string fullpath = path + "UserData.csv.tmp";
+    //string fullpath = path + "UserData.csv.tmp";
```
Comment (typokign) [inline src/datastore/UserDataStore.cpp:105]: cruft

File: src/datastore/UserDataStore.cpp [hunk 31]
```diff
@@ -137,6 +146,32 @@ void UserDataStore::save(const string path)
 
 }
 
+User &UserDataStore::FindbyNumber(int Number)
```
Comment (typokign) [inline src/datastore/UserDataStore.cpp:149]: camelCase

File: src/datastore/UserDataStore.cpp [hunk 32]
```diff
@@ -137,6 +146,32 @@ void UserDataStore::save(const string path)
...
+    {
+        if ( (*it).m_nNumber == Number)
+        {
+            return *it;
+        }
+    }
+    return *(list.end());  // never reached  - should throw exception
+}
+
+bool UserDataStore::DuplicateNumPresent(int Number)
```
Comment (typokign) [inline src/datastore/UserDataStore.cpp:161]: camelCase

File: src/datastore/UserDataStore.cpp [hunk 33]
```diff
@@ -137,6 +146,32 @@ void UserDataStore::save(const string path)
...
+{
+    bool dupe_found = false;
```
Comment (typokign) [inline src/datastore/UserDataStore.cpp:163]: camelCase

**Review by typokign — CHANGES_REQUESTED** [review]
hey thanks lots of work done in this PR - comments below are mainly stylistic. I also think I saw a lot of duplicate code in the file parsing logic - maybe that can be abstracted elsewhere

**Comment by ChrisRzech** [comment]
Recreated by #53.

**Pull request closed without merging**